connector_test.go

```diff
@@ -123,37 +117,40 @@ var volumeConfig = `
 }
 `
 
-func TestSimpleVolume(t *testing.T) {
-	logger := log.NewTestLogger(t)
+func bindMountHelper(t *testing.T, options string) {
+	t.Helper()
 	fileContent, err := os.ReadFile("./tests/volume/test_file.txt")
 	assert.NoError(t, err)
 
-	connector, _ := getConnector(t, volumeConfig)
-	cwd, err := os.Getwd()
-	assert.NoError(t, err)
-	// disable selinux on the test folder in order to make the file readable from within the container
-	cmd := exec.Command("chcon", "-Rt", "svirt_sandbox_file_t", fmt.Sprintf("%s/tests/volume", cwd)) //nolint:gosec
-	err = cmd.Run()
-	if err != nil {
-		logger.Warningf("failed to set SELinux permissions on folder, chcon error: %s, this may cause test failure if SELinux is enabled.", err.Error())
-	}
+	connector, _ := getConnector(t, fmt.Sprintf(volumeConfig, options))
 
 	container, err := connector.Deploy(
 		context.Background(),
 		"quay.io/arcalot/podman-deployer-test-helper:0.1.0")
 	assert.NoError(t, err)
 
-	var containerInput = []byte("volume\n")
-	_, err = container.Write(containerInput)
-	assert.NoError(t, err)
+	t.Cleanup(func() { assert.NoError(t, container.Close()) })
+
+	assert.NoErrorR[int](t)(container.Write([]byte("volume\n")))
```

Comment:
I hate tests like this which rely on someone going off to figure out what `podman-deployer-test-helper` repo and figuring out what the heck the entry point is in order to figure out what the input `volume` is supposed to do. The "validation" code below certainly doesn't help. 😦 

But, fine; the `test_script.sh` will `cat /test/test_file.txt` and we expect ... at least one byte? That's not really much validation. Sure, the old code would have ignored the 3-split entirely and the volume wouldn't be mapped: but beyond the fact that we're not falling down there, what do we know? Not much, given that we expect the exact same behavior for both no volume options and the full set of volume options. Yeah, we can get the file and the container setup didn't fail, sure.

But there's nothing to even suggest that the options weren't simply ignored. We're not proving anything about the labeling of the mapped volume, much less the `ro` and `noexec` behavior... and I think we'd like to be sure all of these options get passed through to `podman` and take effect, wouldn't we?

Reply:
> I hate tests like this which rely on someone going off to figure out what `podman-deployer-test-helper` repo and figuring out what the heck the entry point is in order to figure out what the input `volume` is supposed to do. The "validation" code below certainly doesn't help. 😦

I don't disagree, but I think this is the nature of the beast, here:  we're endeavoring to test container deployment, so, yeah, there are going to be containers involved, and it's gonna be annoying to figure out what they do.  (To be honest, I didn't even look....)

> what do we know? Not much

True.  (I'm afraid that there is lots of room for improvement in the tests.)  The best that I can say is that I did try putting in a garbage option and the test did report a failure.  (The failure report was basically just, "Hey, I got no output!", but it certainly suggests that the non-failure cases actually work.)

> But there's nothing to even suggest that the options weren't simply ignored. We're not proving anything about the labeling of the mapped volume, much less the `ro` and `noexec` behavior... and I think we'd like to be sure all of these options get passed through to `podman` and take effect, wouldn't we?

True...but, at some point we would be crossing over from testing Arca to testing Podman...and, we don't want to do too much of the latter.  (I'm consoling myself here by claiming that I didn't make the test any worse...although, the fact that this PR removes the SELinux stuff makes me a tad suspicious...but only a tad....)

Reply:
Long term, that container should probably be a local container built in the process of running the test suite.

Reply:
Sounds like a reasonable idea in principle; implementation of it might be a bit tricky...but given the other shell commands we run, I suppose we can do a couple more to produce the container image (although we should probably make provisions to remove it when we're done).

But, not in this PR....

Reply:
> True...but, at some point we would be crossing over from testing Arca to testing Podman

Sure, but my concern is not so much that we're not testing that the container does the right thing with `:Z` mapping as with the fact that we're not actually confirming that the label gets to podman at all.

Reply:
Your concerns seem to have been well-founded:  when I enhanced the pass/fail assertion in the test along the lines that you requested, the test no longer passed for the no-options case.  However, once I added the original test's SELinux disablement back in, then all cases pass.

I wouldn't go so far as to say that the test is now "good", but it is "better", and, hopefully, it's "good enough".  To make it "good" would likely require "architectural" changes to the test (and possibly changes to the container image as well).  The problem is that all the cases are mapping the same file into various containers which are potentially running concurrently (so, the difference between `:Z` and `:z` is potentially significant, and I'm not sure what the implications of running with neither are in this situation); also, we're avoiding the SELinux labeling problem on the mapped directory by manually relabeling it in one of the cases, which likely renders the `:Z`/`:z` options moot at least some of the time (depending on the sequence of execution of the cases).

Anyway, the test now passes when I run it locally; as long as that's not just luck and it passes reliably in the CI and for Jared, I'm inclined to leave it as is, at least for now.

Reply:
> as long as that's not just luck and it passes reliably in the CI

Well, either way, it fails in the CI now; but, [I'm not sure it ever actually worked](https://github.com/arcalot/arcaflow-engine-deployer-podman/pull/52#issuecomment-2024279113) there.

So, the test is now correctly failing, at least in the CI, and we need to decide what to do next....